Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_NSException.m

```diff
@@ -68,6 +68,8 @@
         NSArray *addresses = [exception callStackReturnAddresses];
         NSUInteger numFrames = addresses.count;
         uintptr_t *callstack = malloc(numFrames * sizeof(*callstack));
+        assert(callstack != NULL);
```

Comment:
L: is this executed on production code? some languages don't run `asserts` on release mode

Reply:
Yeah assert is used to fail in debug builds and to be explicit about intent only which is fine in this case if I understood the motivation of this PR

Reply:
What @bruno-garcia wrote. There is not much that we can do if malloc fails here.

Reply:
I see, I didn't get the motivation because:

> unblock the customer to use Sentry

how would be a blocker if its only ran on debug builds and they dont have any effects? but yeah maybe their static analysis were complaining a lot, got it.